src/components/nano/NanoContractHistory.js

```diff
@@ -0,0 +1,238 @@
+  const [errorMessage, setErrorMessage] = useState('');
+  // hasBefore {boolean} If 'Previous' button should be enabled 
+  const [hasBefore, setHasBefore] = useState(false);
+  // hasAfter {boolean} If 'Next' button should be enabled 
+  const [hasAfter, setHasAfter] = useState(false);
+
+  /**
+   * useCallback is important here to update this method with new history state
+   * otherwise it would be fixed the moment the event listener is started in the useEffect
+   * with the history as an empty array
+   *
+   * tx {Transaction} Transaction object that arrived from the websocket
+   */
+  const updateListWs = useCallback((tx) => {
+    // We only add to the list if it's the first page and it's a new tx from this nano
+    if (!hasBefore) {
+      if (tx.version === hathorLib.constants.NANO_CONTRACTS_VERSION && tx.nc_id === ncId) {
+        let nanoHistory = [...history];
+        const willHaveAfter = (hasAfter || nanoHistory.length === NANO_CONTRACT_TX_HISTORY_COUNT)
+        // This updates the list with the new element at first
+        nanoHistory = helpers.updateListWs(nanoHistory, tx, NANO_CONTRACT_TX_HISTORY_COUNT);
+
+        // Now update the history
+        setHistory(nanoHistory);
+        setHasAfter(willHaveAfter);
+      }
+    }
+  }, [history, hasAfter, hasBefore, ncId]);
+
+  /**
+   * useCallback is needed here because this method is used as a dependency in the useEffect
+   *
+   * after {string | null} Hash to use for pagination when user clicks to fetch the next page
+   * before {string | null} Hash to use for pagination when user clicks to fetch the previous page
+   */
+  const loadData = useCallback(async (after, before) => {
+    try {
+      const data = await nanoApi.getHistory(ncId, NANO_CONTRACT_TX_HISTORY_COUNT, after, before);
+      if (before) {
+        // When we are querying the previous set of transactions
+        // the API return the oldest first, so we need to revert the history
+        reverse(data.history);
+      }
+      setHistory(data.history);
+
+      if (!after && !before) {
+        // This is the first load without query params, so if has_more === true
+        // we must enable next button
+        setHasAfter(data.has_more);
+        setHasBefore(false);
+        return;
+      }
+
+      if (after) {
+        // We clicked the next button, so we have before page
+        // and we will have the next page if has_more === true
+        setHasAfter(data.has_more);
+        setHasBefore(true);
+        return;
+      }
+
+      if (before) {
+        // We clicked the previous button, so we have next page
+        // and we will have the previous page if has_more === true
+        setHasAfter(true);
+        setHasBefore(data.has_more);
+        if (!data.has_more) {
+          // We are in the first page and clicked the Previous button
+          // so we must clear the query params
+          pagination.clearOptionalQueryParams();
```

Comment:
:star: Nice behavior here! Maybe later we could find a way to centralize this on the Pagination util so that other screens also inherit this same pattern without having to explicitly code it on the screen.